server/prisma/schema.prisma

```diff
@@ -200,13 +200,30 @@ model Group {
   memberActivity MemberActivity[]
   patronages     Patron[]
   socialLinks    GroupSocialLinks[]
+  roleOrder      GroupRoleOrder[]
 
   // Constraints
   @@unique([name], map: "groups_name_key")
   // Map
   @@map("groups")
 }
 
+model GroupRoleOrder {
+  groupId Int
+  role    GroupRole @unique @default(member)
+  index   Int       @unique
+
+  // Relations
+  group Group @relation(fields: [groupId], references: [id], onDelete: Cascade, onUpdate: NoAction)
+
+  // Constraints
+  @@id([groupId])
```

Comment:
The ID should be a compound key of [groupId, role, index], and the `@unique` tags should be removed from those fields, otherwise there could only be 1 row per groupId, and 1 row per role. See `Membership` as an example.

Reply:
yup, sorry hadn't updated this, realized that in my tests